Bottom: b7de5ca8b0326ee7422505b5ac00f365e53e8286
Top:    98b0661f7b74d4b77513fb0b1e12b4fcf235c725
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-08-26 03:12:38 +0100

Refresh of gdl

---

diff --git a/configure.ac b/configure.ac
index 7628e9d..5b0b6e7 100644
--- a/configure.ac
+++ b/configure.ac
@@ -403,6 +403,14 @@ AC_ARG_ENABLE([gl],
 AC_MSG_RESULT([$enable_gl])
 AM_CONDITIONAL(USE_GL, test x$enable_gl = xyes)
 
+AC_MSG_CHECKING([for whether to use GDL library])
+AC_ARG_ENABLE([gdl],
+[  --enable-gdl           Enable GDL library (with GTK HID)],
+[],[enable_gdl=no])
+AC_MSG_RESULT([$enable_gdl])
+AM_CONDITIONAL(USE_GDL, test x$enable_gdl = xyes)
+
+
 if test "x$enable_gl" = "xyes"; then
 	case " $with_gui " in
 		*\ gtk\ *) ;;
@@ -784,6 +792,16 @@ $GTKGLEXT_PKG_ERRORS])]
 	GTKGLEXT_VER=`$PKG_CONFIG gtkglext-1.0 --modversion`
 	fi
 
+	if test "x$enable_gdl" = "xyes"; then
+		# Check for GDL
+		PKG_CHECK_MODULES(GDL, gdl-1.0, , [AC_MSG_ERROR([
+*** Required version of libgdl (gnome dock library) is not installed - please install first ***
+Please review the following errors:
+$GDL_PKG_ERRORS])]
+		)
+	GDL_VER=`$PKG_CONFIG gdl-1.0 --modversion`
+	fi
+
 	;;
 
       gcode|nelma|png )
@@ -1071,8 +1089,8 @@ AM_CONDITIONAL(DEBUG_BUILD, test x$enable_debug = xyes)
 
 # ------------- Complete set of CFLAGS and LIBS -------------------
 
-CFLAGS="$CFLAGS $X_CFLAGS $DBUS_CFLAGS $GLIB_CFLAGS $GTK_CFLAGS $CAIRO_CFLAGS $GTKGLEXT_CFLAGS"
-LIBS="$LIBS $XM_LIBS $DBUS_LIBS $X_LIBS $GLIB_LIBS $GTK_LIBS $DMALLOC_LIBS $GD_LIBS $INTLLIBS $CAIRO_LIBS $GTKGLEXT_LIBS"
+CFLAGS="$CFLAGS $X_CFLAGS $DBUS_CFLAGS $GLIB_CFLAGS $GTK_CFLAGS $CAIRO_CFLAGS $GTKGLEXT_CFLAGS $GDL_CFLAGS"
+LIBS="$LIBS $XM_LIBS $DBUS_LIBS $X_LIBS $GLIB_LIBS $GTK_LIBS $DMALLOC_LIBS $GD_LIBS $INTLLIBS $CAIRO_LIBS $GTKGLEXT_LIBS $GDL_LIBS"
 
 
 # if we have gcc then add -Wall
diff --git a/src/hid/gtk/gui-top-window.c b/src/hid/gtk/gui-top-window.c
index 16c26d2..78274fa 100644
--- a/src/hid/gtk/gui-top-window.c
+++ b/src/hid/gtk/gui-top-window.c
@@ -88,6 +88,8 @@ a zoom in/out.
 #include "../hidint.h"
 #include "hid/common/hid_resource.h"
 
+#include <gdl/gdl.h>
+
 #include "action.h"
 #include "autoplace.h"
 #include "autoroute.h"
@@ -2157,6 +2159,10 @@ ghid_build_pcb_top_window (void)
   GtkWidget *viewport, *ebox, *vbox, *frame;
   GtkWidget *label;
   GtkWidget *trackball;
+
+  GtkWidget *dock;
+  GtkWidget *tmp;
+
   GHidPort *port = &ghid_port;
   gchar *s;
   GtkWidget *scrolled;
@@ -2246,11 +2252,30 @@ ghid_build_pcb_top_window (void)
   hbox_middle = gtk_hbox_new (FALSE, 0);
   gtk_box_pack_start (GTK_BOX (vbox_main), hbox_middle, TRUE, TRUE, 3);
 
+  dock = gdl_dock_new ();
+  gtk_box_pack_start (GTK_BOX (hbox_middle), dock, FALSE, FALSE, 0);
+  gtk_widget_show (dock);
 
+  tmp = gdl_dock_bar_new (GDL_DOCK (dock));
+  gdl_dock_bar_set_orientation (GDL_DOCK_BAR (dock), GTK_ORIENTATION_VERTICAL);
+  gtk_box_pack_start (GTK_BOX (hbox_middle), dock, FALSE, FALSE, 0);
+  gtk_widget_show (dock);
+
+#if 1
   /* -- Left control bar */
   ebox = gtk_event_box_new ();
   gtk_widget_set_events (ebox, GDK_EXPOSURE_MASK);
-  gtk_box_pack_start (GTK_BOX (hbox_middle), ebox, FALSE, FALSE, 3);
+
+  tmp = gdl_dock_item_new ("Left bar", "Left bar : description",
+                           GDL_DOCK_ITEM_BEH_NEVER_HORIZONTAL |
+                           GDL_DOCK_ITEM_BEH_CANT_DOCK_TOP |
+                           GDL_DOCK_ITEM_BEH_CANT_DOCK_BOTTOM |
+                           GDL_DOCK_ITEM_BEH_CANT_CLOSE);
+  gtk_container_add (GTK_CONTAINER (tmp), ebox);
+  gdl_dock_add_item (GDL_DOCK (dock), GDL_DOCK_ITEM (tmp), GDL_DOCK_LEFT);
+  gtk_widget_set_size_request (dock, 5, -1);
+  gtk_widget_set_size_request (dock, -1, -1);
+  gtk_widget_show (tmp);
 
   /* 
    * This box will also be made insensitive when the gui needs
@@ -2293,6 +2318,7 @@ ghid_build_pcb_top_window (void)
   vbox = gtk_vbox_new(FALSE, 0);
   gtk_box_pack_start(GTK_BOX(hbox), vbox, FALSE, FALSE, 4);
   make_route_style_buttons(vbox, port);
+#endif
 
   vbox = gtk_vbox_new (FALSE, 0);
   gtk_box_pack_start (GTK_BOX (hbox_middle), vbox, TRUE, TRUE, 0);
